### PR TITLE
Implement cpu binding feature in CCompute

### PR DESCRIPTION
**Pull request by XiaowenJiang**
Signed-off-by: Julie Jiang <julie.jiang@emc.com>

### CODE REVIEW

**Comment by XiaowenJiang** [comment]
@weia5-emc  Please help us to take a look at the implementation

**Comment by emcinfrasim** [comment]
<details><summary>Test Error Details</summary><p>
TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 170, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 172, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 E272 
<br>ErrorMessage: line 182, E272 multiple spaces before keyword 
<br> 
<br>TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 184, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 188, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 E226 
<br>ErrorMessage: line 553, E226 missing whitespace around arithmetic operator 
<br> 
<br>TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 E231 
<br>ErrorMessage: line 564, E231 missing whitespace after ':' 
<br> 
<br></p></details>

**Comment by emcinfrasim** [comment]
<details><summary>Test Error Details</summary><p>
TestName: test_peer_bmc_fru 
<br>ErrorType: exceptions.TypeError 
<br>ErrorMessage: line None, argument of type 'NoneType' is not iterable
-------------------- >> begin captured stdout << ---------------------
Connecting to SSH server: root@192.168.188.91:8022
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Connection failed for executing command sudo ipmitool fru print 0.
Connecting to SSH server: root@192.168.188.91:8022
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Connection failed for executing command sudo ipmitool -t 0x1e fru print 0.

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
nose_cov: DEBUG: nose-cov begin
ivn.core: INFO: [Validate topology]
ivn.core: INFO: All namespaces in index have definition.
ivn.core: INFO: All openvswitch in index have definition.
ivn.core: INFO: Namespace node0ns interfaces and bridges config is legal.
ivn.core: INFO: Namespace node1ns interfaces and bridges config is legal.
ivn.core: INFO: Openvswitch br-int ports config is legal.
ivn.core: INFO: Connection from namespace interface ns0-einf0 to openvswitch port vint0 is defined.
ivn.core: INFO: Connection from namespace interface ns1-einf0 to openvswitch port vint1 is defined.
ivn.core: INFO: Topology pass validation.
ivn.core: INFO: [Define openvswitches]
ivn.core: INFO: vswitch br-int is created.
ivn.core: INFO: port vint0 is added to br-int.
ivn.core: INFO: port vint1 is added to br-int.
ivn.core: INFO: Openvswitch br-int is defined in /etc/network/interfaces.d and restarted.
ivn.core: INFO: [Define namespaces]
ivn.core: INFO: namespace node1ns is created.
ivn.core: INFO: Setting Id 1 for namespace node1ns.
ivn.core: INFO: Validation of creating namespace node1ns passed.
ivn.core: INFO: interface ns1-einf0 in namespace node1ns is created, peer: vint1.
ivn.core: INFO: bridge br0 in namespace node1ns is created on interface ns1-einf0.
ivn.core: INFO: namespace node0ns is created.
ivn.core: INFO: Setting Id 2 for namespace node0ns.
ivn.core: INFO: Validation of creating namespace node0ns passed.
ivn.core: INFO: interface ns0-einf0 in namespace node0ns is created, peer: vint0.
ivn.core: INFO: bridge br0 in namespace node0ns is created on interface ns0-einf0.
ivn.core: INFO: [Set openvswitch ports up]
ivn.core: INFO: set openvswitch br-int up.
ivn.core: INFO: set port vint0 up.
ivn.core: INFO: set port vint1 up.
ivn.core: INFO: [Set namespace interfaces up]
ivn.core: INFO: namespace node1ns interfaces are defined in /etc/netns/node1ns/network.
ivn.core: INFO: interfaces in namespace node1ns are restarted.
ivn.core: INFO: namespace node0ns interfaces are defined in /etc/netns/node0ns/network.
ivn.core: INFO: interfaces in namespace node0ns are restarted.
ivn.core: INFO: [Setup portforward]
ivn.core: INFO: forwarding from 192.168.188.91:5901 to host:15901
ivn.core: INFO: forwarding from 192.168.188.92:5901 to host:25901
ivn.core: INFO: forwarding from 192.168.188.91:8022 to host:18022
ivn.core: INFO: forwarding from 192.168.188.92:8022 to host:28022
1Model: INFO: [compute] infrasim can't find numactl in this environment
1Model: INFO: test0-socat's command line: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test0/.pty0_test0,waitslave,unlink-close=0 unix-listen:/home/infrasim/.infrasim/test0/.serial,fork
1Model: INFO: [ 19352  ] test0-socat starts to run
1Model: INFO: test0-bmc's command line: ipmi_sim -c /home/infrasim/.infrasim/test0/etc/vbmc.conf -f /home/infrasim/.infrasim/test0/data/dell_r730.emu -n -s /home/infrasim/.infrasim/test0/data -l /var/log/infrasim/test0/ipmi_sim.log
1Model: INFO: test0-racadm's command line: racadmsim test0 0.0.0.0 10022 admin admin auto
1Model: INFO: [ 19358  ] test0-racadm starts to run
1Model: INFO: test0-monitor's command line: infrasim-monitor test0 0.0.0.0 9005
1Model: INFO: [ 19362  ] test0-monitor starts to run
2Model: INFO: [compute] infrasim can't find numactl in this environment
2Model: INFO: test1-socat's command line: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test1/.pty0_test1,waitslave,unlink-close=0 unix-listen:/home/infrasim/.infrasim/test1/.serial,fork
2Model: INFO: [ 19370  ] test1-socat starts to run
2Model: INFO: test1-bmc's command line: ipmi_sim -c /home/infrasim/.infrasim/test1/etc/vbmc.conf -f /home/infrasim/.infrasim/test1/data/quanta_d51.emu -n -s /home/infrasim/.infrasim/test1/data -l /var/log/infrasim/test1/ipmi_sim.log
2Model: INFO: test1-monitor's command line: infrasim-monitor test1 0.0.0.0 9005
2Model: INFO: [ 19376  ] test1-monitor starts to run
--------------------- >> end captured logging << --------------------- 
<br> 
<br>TestName: test_peer_bmc_lan 
<br>ErrorType: exceptions.TypeError 
<br>ErrorMessage: line None, argument of type 'NoneType' is not iterable
-------------------- >> begin captured stdout << ---------------------
Connecting to SSH server: root@192.168.188.91:8022
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Connection failed for executing command sudo ipmitool lan print.
Connecting to SSH server: root@192.168.188.91:8022
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Connection failed for executing command sudo ipmitool -t 0x1e lan print.

--------------------- >> end captured stdout << ---------------------- 
<br> 
<br>TestName: test_peer_bmc_sel 
<br>ErrorType: exceptions.AssertionError 
<br>ErrorMessage: line None, peer_bmc can not Obtain sensor
-------------------- >> begin captured stdout << ---------------------
Connecting to SSH server: root@192.168.188.91:8022
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Connection failed for executing command sudo ipmitool sel list.
Connecting to SSH server: root@192.168.188.91:8022
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Connection failed for executing command sudo ipmitool -t 0x1e sel list.

--------------------- >> end captured stdout << ---------------------- 
<br> 
<br>TestName: test_peer_bmc_sensor 
<br>ErrorType: exceptions.AssertionError 
<br>ErrorMessage: line None, peer_bmc can not Obtain sensor
-------------------- >> begin captured stdout << ---------------------
Connecting to SSH server: root@192.168.188.91:8022
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Connection failed for executing command sudo ipmitool sensor list.
Connecting to SSH server: root@192.168.188.91:8022
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Failed to connect 192.168.188.91: [Errno None] Unable to connect to port 8022 on 192.168.188.91
Connection failed for executing command sudo ipmitool -t 0x1e sensor list.

--------------------- >> end captured stdout << ---------------------- 
<br> 
<br>TestName: test_peer_node_cycle 
<br>ErrorType: exceptions.AssertionError 
<br>ErrorMessage: line None, 
-------------------- >> begin captured stdout << ---------------------
Connecting to SSH server: root@192.168.188.92:8022
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Connection failed for executing command ls.

--------------------- >> end captured stdout << ---------------------- 
<br> 
<br>TestName: test_peer_node_off_on 
<br>ErrorType: exceptions.AssertionError 
<br>ErrorMessage: line None, 
-------------------- >> begin captured stdout << ---------------------
Connecting to SSH server: root@192.168.188.92:8022
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Failed to connect 192.168.188.92: [Errno None] Unable to connect to port 8022 on 192.168.188.92
Connection failed for executing command ls.

--------------------- >> end captured stdout << ---------------------- 
<br> 
<br>TestName: test_set_bmc_lan_channel 
<br>ErrorType: exceptions.AssertionError 
<br>ErrorMessage: line None, 
-------------------- >> begin captured stdout << ---------------------
?[37m[?[0m ?[32m21523 ?[0m ?[37m]?[0m test-socat is running.
?[37m[?[0m ?[32m21526 ?[0m ?[37m]?[0m test-bmc is running.
?[37m[?[0m ?[32m21533 ?[0m ?[37m]?[0m test-node is running.
?[37m[?[0m ?[32m21543 ?[0m ?[37m]?[0m test-racadm is running.
?[37m[?[0m ?[32m21549 ?[0m ?[37m]?[0m test-monitor is running.

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
3Model: INFO: [compute] infrasim can't find numactl in this environment
3Model: INFO: [BaseDrive] Creating drive: /home/infrasim/.infrasim/test/disk-sata-0-0.img
3Model: INFO: test-socat's command line: /usr/bin/socat pty,link=/home/infrasim/.infrasim/test/.pty0_test,waitslave,unlink-close=0 unix-listen:/home/infrasim/.infrasim/test/.serial,fork
3Model: INFO: [ 21523  ] test-socat starts to run
3Model: INFO: test-bmc's command line: ipmi_sim -c /home/infrasim/.infrasim/test/etc/vbmc.conf -f /home/infrasim/.infrasim/test/data/dell_r730.emu -n -s /home/infrasim/.infrasim/test/data -l /var/log/infrasim/test/ipmi_sim.log
3Model: INFO: [ 21526  ] test-bmc starts to run
3Model: INFO: [ 21533  ] test-node is running
3Model: INFO: test-racadm's command line: racadmsim test 0.0.0.0 10022 admin admin auto
3Model: INFO: [ 21543  ] test-racadm starts to run
3Model: INFO: test-monitor's command line: infrasim-monitor test 0.0.0.0 9005
3Model: INFO: [ 21549  ] test-monitor starts to run
--------------------- >> end captured logging << --------------------- 
<br> 
<br></p></details>

File: infrasim/model/core/task.py [hunk 1]
```diff
@@ -140,6 +140,8 @@ def run(self):
                     self.__print_task(self.get_task_pid(), self.__task_name, "running")
                     self.__logger.info("[ {:<6} ] {} is running".format(self.get_task_pid(),
                                                                         self.__task_name))
+                    if "node" in self.__task_name:
```
Comment (weia5-emc) [inline infrasim/model/core/task.py]: it leads to failure (no attribute) if user use "***node***" as node name in yaml.
suggest using an empty (abstract) function in task (base class) and implementing a concrete function in CCompute (subclass), which handles the actual process.

**Comment by weia5-emc** [comment]
Suggestion on binding policy:
Separate physical CPU core into groups for different purpose, for example.  some cores will be used for vCPU, some will be used for qemu main loop, and IO thread, etc.... so that they can be treated differently by priority.

**Comment by XiaowenJiang** [comment]
@weia5-emc We've modified the implementation for more explicit binding type and move the class into CElement, please help us to review the changes.

**Comment by emcinfrasim** [comment]
<details><summary>Test Error Details</summary><p>
TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E302 
<br>ErrorMessage: line 14, E302 expected 2 blank lines, found 1 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E265 
<br>ErrorMessage: line 23, E265 block comment should start with '# ' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E265 
<br>ErrorMessage: line 28, E265 block comment should start with '# ' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E265 
<br>ErrorMessage: line 38, E265 block comment should start with '# ' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 48, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 50, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E272 
<br>ErrorMessage: line 60, E272 multiple spaces before keyword 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 62, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 66, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 103, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 105, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 113, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 115, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E226 
<br>ErrorMessage: line 155, E226 missing whitespace around arithmetic operator 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E231 
<br>ErrorMessage: line 166, E231 missing whitespace after ':' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E265 
<br>ErrorMessage: line 183, E265 block comment should start with '# ' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E303 
<br>ErrorMessage: line 187, E303 too many blank lines (2) 
<br> 
<br>TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 F401 
<br>ErrorMessage: line 10, F401 're' imported but unused 
<br> 
<br>TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 F401 
<br>ErrorMessage: line 11, F401 'multiprocessing' imported but unused 
<br> 
<br></p></details>

File: infrasim/model/core/task.py [hunk 2]
```diff
@@ -242,3 +246,6 @@ def execute_command(command, logger, log_path="", duration=1):
             raise CommandRunFailed(command, errout)
 
         return proc.pid
+
+    def bind_cpus(self):
```
Comment (weia5-emc) [inline infrasim/model/core/task.py]: 👏👏👏

**Comment by emcinfrasim** [comment]
<details><summary>Test Error Details</summary><p>
TestName: infrasim/model/tasks/compute.py 
<br>ErrorType: flake8 F401 
<br>ErrorMessage: line 10, F401 'multiprocessing' imported but unused 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E265 
<br>ErrorMessage: line 23, E265 block comment should start with '# ' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E265 
<br>ErrorMessage: line 28, E265 block comment should start with '# ' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E265 
<br>ErrorMessage: line 38, E265 block comment should start with '# ' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 48, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 50, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E272 
<br>ErrorMessage: line 60, E272 multiple spaces before keyword 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 62, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 66, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 103, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 105, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 113, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E502 
<br>ErrorMessage: line 115, E502 the backslash is redundant between brackets 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E226 
<br>ErrorMessage: line 155, E226 missing whitespace around arithmetic operator 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E231 
<br>ErrorMessage: line 166, E231 missing whitespace after ':' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E265 
<br>ErrorMessage: line 183, E265 block comment should start with '# ' 
<br> 
<br>TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E303 
<br>ErrorMessage: line 187, E303 too many blank lines (2) 
<br> 
<br></p></details>

**Comment by emcinfrasim** [comment]
<details><summary>Test Error Details</summary><p>
TestName: infrasim/model/elements/cpu_binding.py 
<br>ErrorType: flake8 E265 
<br>ErrorMessage: line 183, E265 block comment should start with '# ' 
<br> 
<br></p></details>

File: infrasim/model/core/task.py [hunk 3]
```diff
@@ -73,6 +73,9 @@ def get_workspace(self):
     def set_log_path(self, log_path):
         self.__log_path = log_path
 
+    def bind_cpus_with_policy(self):
```
Comment (weia5-emc) [inline infrasim/model/core/task.py]: why is this function here?

**Comment by fub2** [comment]
Good team work!!! @XiaowenJiang @bcyan @weia5-emc

**Pull request merged**